File: python/maxSubArray.py
def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxValue = nums[0]
        current  = 0
        for item in nums:
            if current < 0:
                current  = 0
            
            
            current += item
            
            maxValue = max(current, maxValue)
        return maxValue

File: swift/maxSubArray.swift
func maxSubArray(_ nums: [Int]) -> Int {
        
        var finalSum:Int = nums[0]
        var currentSum:Int = 0
        for item in nums {
            
            if currentSum <= 0 {
                currentSum = 0
            }
            currentSum = currentSum + item
            
            if currentSum > finalSum {
                finalSum = currentSum
            }
        }
        
        return finalSum
    }
